opengrok-indexer/src/main/java/org/opengrok/indexer/history/FileHistoryCache.java

```diff
@@ -103,38 +103,32 @@ private void doFileHistory(String filename, List<HistoryEntry> historyEntries,
             Repository repository, File srcFile, File root, boolean renamed)
             throws HistoryException {
 
-        History hist = null;
+        File file = new File(root, filename);
+        // Only store directory history for the top-level directory.
+        if (file.isDirectory() && !filename.equals(repository.getDirectoryName())) {
+            LOGGER.log(Level.FINE, "Not storing history cache for {0}", file);
```

Comment:
a reason in the message could be helpful for debugging

Reply:
sure, fixed